app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt

```diff
@@ -224,18 +227,37 @@ class ZimHostActivity : BaseActivity(), ZimHostCallbacks, ZimHostContract.View {
   // Advice user to turn on hotspot manually for API<26
   private fun startHotspotManuallyDialog() {
 
-    alertDialogShower.show(KiwixDialog.StartHotspotManually,
-      ::launchTetheringSettingsScreen,
-      { startActivity(Intent(Settings.ACTION_WIFI_SETTINGS)) },
-      {
-        progressDialog = ProgressDialog.show(
-          this,
-          getString(R.string.progress_dialog_starting_server), "",
-          true
-        )
-        startService(createHotspotIntent(ACTION_CHECK_IP_ADDRESS))
-      }
+    val wifiManager =
+      applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
+    when {
+      wifiManager.isWifiEnabled -> alertDialogShower.show(
+        KiwixDialog.WifiOn,
+        { startActivity(Intent(Settings.ACTION_WIFI_SETTINGS)) },
+        {},
+        ::startServer
+      )
+      isHotspotOn(wifiManager) -> startServer()
+      else -> alertDialogShower.show(
+        KiwixDialog.StartHotspotManually,
+        ::launchTetheringSettingsScreen,
+        { startActivity(Intent(Settings.ACTION_WIFI_SETTINGS)) }
+      )
+    }
+  }
+
+  private fun startServer() {
+    progressDialog = ProgressDialog.show(
+      this,
+      getString(R.string.progress_dialog_starting_server), "",
+      true
     )
+    startService(createHotspotIntent(ACTION_CHECK_IP_ADDRESS))
+  }
+
+  private fun isHotspotOn(wifiManager: WifiManager): Boolean {
+    val method: Method = wifiManager.javaClass.getDeclaredMethod("isWifiApEnabled")
```

Comment:
this can throw a lot of exceptions, reflection is innately dangerous and only to be used as a last resort

Reply:
@macgills I went through a few solutions I found in other stack overflow answers, but nothing could get to work. Could you please tell if some method other than reflection can be used?

Reply:
I don't think it can be but wrap it in a try catch

Reply:
Did this.